plugins/amazonq/chat/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevControllerTest.kt

```diff
@@ -96,12 +153,248 @@ class FeatureDevControllerTest : FeatureDevTestBase() {
 
             controller.processPromptChatMessage(message)
 
-            verify(spySession, times(1)).preloader(any(), any())
-            verify(spySession, times(1)).send(any())
+            mockitoVerify(spySession, times(1)).preloader(any(), any())
+            mockitoVerify(spySession, times(1)).send(any())
             assertThat(spySession.send(userMessage).content?.trim()).isEqualTo(exampleGenerateTaskAssistPlanResult.approach)
             assertThat(spySession.send(userMessage).interactionSucceeded).isTrue()
         }
-        verify(authController, times(1)).getAuthNeededStates(any())
+        mockitoVerify(authController, times(1)).getAuthNeededStates(any())
         assertThat(spySession.sessionState).isInstanceOf(RefinementState::class.java)
     }
+
+    @Test
+    fun `test newTask and closeSession followUp`() {
+        /*
+            Testing both followups together as they share logic, atm they could be verified together.
+         */
+        val followUp = FollowUp(FollowUpTypes.NEW_TASK, pillText = "Work on new task")
+        val message = IncomingFeatureDevMessage.FollowupClicked(followUp, testTabId, "", "test-command")
+
+        whenever(featureDevClient.createTaskAssistConversation()).thenReturn(exampleCreateTaskAssistConversationResponse)
+        whenever(chatSessionStorage.getSession(any(), any())).thenReturn(spySession)
+        doNothing().`when`(chatSessionStorage).deleteSession(any())
+
+        mockkObject(AmazonqTelemetry)
+        every { AmazonqTelemetry.endChat(amazonqConversationId = any(), amazonqEndOfTheConversationLatency = any()) } just runs
+
+        runTest {
+            spySession.preloader(userMessage, messenger)
```

Comment:
This test is for controller; so do we need to invoke `spySession.preloader` here?

Reply:
I do becuase the conversationId needs to be exsiting for the piece of code not to fail.